### PR TITLE
Create nvim_get_var binding

### PR DESCRIPTION
**Pull request by shift-d**

### CODE REVIEW

**Comment by shift-d** [comment]
This is my first actual code PR, and I wonder whether every binding works like this. If yes, then I will do more big PRs later.

File: nvim-oxi/src/api/global.rs [hunk 1]
```diff
@@ -116,7 +116,15 @@ pub fn get_mode() -> Dictionary {
...
 
-// get_var
+/// Binding to `nvim_get_var`.
+pub fn get_var<Value>(name: &str) -> Result<Value>
+where
+    Value: FromObject,
+{
+    let mut err = NvimError::new();
+    let obj = unsafe { nvim_get_var(name.into(), &mut err) };
+    err.into_err_or_flatten(|| Value::from_obj(obj))
```
Comment (noib3) [inline nvim-oxi/src/api/global.rs:126]: Just a nitpick: I prefer using `foo.into()` as opposed to `Foo::from(foo)` whenever the compiler can infer the type, as it reads from left to right like natural language. Closures are an exception, where `.map(Foo::from)` > `.map(|foo| foo.into())`.

**Comment by noib3** [comment]
> I wonder whether every binding works like this

Yes, mostly. There are a few conventions to follow, e.g. if the Lua api expects an `opts` table we write an `*Opts` struct deriving `Builder`, and some fields have to be generic over `FromObject`/`ToObject`.

Looking at `buffer.rs` is a great way to see how the Rust api will be structured structured.

File: nvim-oxi/src/api/global.rs (hunk 1 above)
Comment (shift-d) [reply nvim-oxi/src/api/global.rs:126]: I'm sorry, but I don't quite get which part of the code you would suggest to change 😅

Comment (noib3) [reply nvim-oxi/src/api/global.rs:126]: Oops my bad, I read that as `Value::from` (from the `From` trait) instead of `Value::from_obj`.

**Pull request merged**